app/models/scheduled_message.rb

```diff
@@ -6,4 +6,16 @@ class ScheduledMessage < ActiveRecord::Base
   validates :title, presence: true
 
   acts_as_taggable
+
+  def self.filter(params)
+
+    results = self.all.where("lower(title) like ?", "%#{params[:title].downcase}%")
+    results = results.where("lower(body) like ?", "%#{params[:body].downcase}%")
+
+    if !params[:tag].blank?
+      tags = params[:tag].split(",").each { |t| t.strip }
+      results = results.tagged_with(tags, :any => true)
+    end
+    return results
```

Comment:
This method can be re-written along the lines of what I suggested in `sent_scheduled_message.rb`